### PR TITLE
Use Glean Dictionary to search for probes/metrics (fixes #1678)

### LINKED ISSUE #1678: Replace probe search service usage with Glean Dictionary
We should consider using the Glean Dictionary for searching for metric data instead of the probe search service. This would have a number of advantages:

* Simplify our cloud infrastructure (would no longer need to maintain a seperate web service)
* Simplify development (the Glean Dictionary is easier to hack on)

Depends on:

* https://github.com/mozilla/glean-dictionary/pull/834 (for Fenix)
* https://github.com/mozilla/glean-dictionary/issues/926 (for Firefox Desktop)

### PR DESCRIPTION
**Pull request by wlach**
We make use of a new netlify-function based approach in the Glean
Dictionary to replace calls to the probe search service
(https://github.com/mozilla/probe-search). This service is more
straightforward to extend and maintain -- both in terms of code and
operational complexity.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -33,8 +23,8 @@ export default {
     replace({
       __BASE_DOMAIN__: production ? '' : 'http://localhost:8000',
       __GA_TRACKING_ID__: process.env.GA_TRACKING_ID,
-      __BASE_SEARCH_DOMAIN__: SEARCH_DOMAIN,
-      __GLEAN_DICTIONARY_DOMAIN__: 'https://dictionary.telemetry.mozilla.org',
+      __GLEAN_DICTIONARY_DOMAIN__:
+        'https://deploy-preview-834--glean-dictionary-dev.netlify.app',
```
Comment (wlach) [inline rollup.config.js]: Needs to be replaced once the relevant code in the glean dictionary lands (pending).

**Comment by wlach** [comment]
We want to test actual metric pages *as well as* search for firefox (since the metric metadata is now also fetched from the glean dictionary).

Here are a few test cases and the search terms to find them:

* expired metrics: https://glam.telemetry.mozilla.org/firefox/probe/a11y_uia_detection_timing_ms/explore ("uia detection")
* int scalar: https://glam.telemetry.mozilla.org/firefox/probe/a11y_hcm_background/explore?process=parent ("a11y hcm")
* keyed histograms: http://glam.telemetry.mozilla.org/firefox/probe/application_reputation_binary_archive/explore?activeBuckets=%5B%22OtherBinaryFile%22%2C%22ZipFile%22%2C%22RarFile%22%2C%22DmgFile%22%5D&process=parent ("reputation binary archive")
* standard histogram: https://glam.telemetry.mozilla.org/firefox/probe/urlclassifier_lookup_time_2/explore?process=parent ("url classifier lookup")

**Comment by wlach** [comment]
Ok I think this is ready for review now. No rush.

**Review by Iinh — APPROVED** [review]
LGTM! With this change I can see that search is now faster than before 🎉

**Pull request merged**